src/mappings/helpers.ts

```diff
@@ -577,3 +582,78 @@ function createGraphAccountName(
 export function joinID(pieces: Array<string>): string {
   return pieces.join('-')
 }
+
+function min(a: BigDecimal, b: BigDecimal): BigDecimal {
+  return a < b ? a : b
+}
+
+function max(a: BigDecimal, b: BigDecimal): BigDecimal {
+  return a > b ? a : b
+}
+
+export function calculateOwnStakeRatio(indexer: Indexer): BigDecimal {
+  let graphNetwork = GraphNetwork.load('1')
+  let delegationRatio = BigInt.fromI32(graphNetwork.delegationRatio)
+  let maxPossibleTotalUsable =
+    indexer.stakedTokens.toBigDecimal() +
+    (indexer.stakedTokens.toBigDecimal() * delegationRatio.toBigDecimal())
+  let currentTotalStake =
+    indexer.stakedTokens.toBigDecimal() + indexer.delegatedTokens.toBigDecimal()
+  let totalUsable = min(maxPossibleTotalUsable, currentTotalStake)
+  return totalUsable == BigDecimal.fromString('0')
+    ? BigDecimal.fromString('0')
+    : indexer.stakedTokens.toBigDecimal() / totalUsable
+}
+
+export function calculateDelegatedStakeRatio(indexer: Indexer): BigDecimal {
+  let graphNetwork = GraphNetwork.load('1')
+  let delegationRatio = BigInt.fromI32(graphNetwork.delegationRatio)
```

Comment:
Yeah, the only way to prevent this is to use the "Allocated stake based" formula, but I'm not sure that the resulting effective cut will be relevant for delegators with that formula.

As a point of reference, the allocation based formula would show higher effective cut % depending on the amount of unallocated stake that the indexer has, because the ratios would skyrocket as a result (since total usable, delegated and own stake would be far higher than the allocated stake when you have high unallocated stake amounts).

Although, as you mentioned, it should only really affect over-delegated indexers, and it's updated quite often (unlike the delegation ratio), so it shouldn't display "bad" data often

Reply:
My gut feeling is we should go with what you have implemented here, get some feedback from users, and see how clear it is to them. And then maybe we can change it. 

I wonder if we can get Nena to use it on the front end, before officially releasing an update to everyone, so that maybe we can get feedback and change it